iotedgedev/modules.py

```diff
@@ -36,10 +41,24 @@ def add(self, name, template):
         elif os.path.exists(os.path.join(cwd, name)):
             raise ValueError("Module \"{0}\" already exists under {1}".format(name, os.path.abspath(self.envvars.MODULES_PATH)))
 
-        deployment_manifest = DeploymentManifest(self.envvars, self.output, self.utility, self.envvars.DEPLOYMENT_CONFIG_TEMPLATE_FILE, True)
-
         repo = "{0}/{1}".format("${CONTAINER_REGISTRY_SERVER}", name.lower())
-        if template == "csharp":
+        if template == "c":
+            github_prefix = "https://github.com/Azure"
+            git_repo = "azure-iot-edge-c-module"
+            branch = "master"
+            url = "{0}/{1}/archive/{2}.zip".format(github_prefix, git_repo, branch)
+            response = urlopen(url)
```

Comment:
Thoughts on caching this zip locally?  Have you tested the download time of this zip file?  Does it get all recursive projects? Do we need those projects?

Reply:
The size of the zip file is only 8KB, so we'd better always fetch the latest instead of caching them. The dependencies are installed via `apt-get` in the [Dockerfile](https://github.com/Azure/azure-iot-edge-c-module/blob/master/Dockerfile.amd64#L6)